src/data/finsupp/basic.lean

```diff
@@ -1483,6 +1483,50 @@ end
 
 end comap_domain
 
+section option
+
+/-- Restrict a finitely supported function on `option α` to a finitely support function on `α`. -/
+def some [has_zero M] (f : option α →₀ M) : α →₀ M :=
+f.comap_domain option.some (by tidy)
```

Comment:
```suggestion
f.comap_domain option.some (λ _, by simp)
```

otherwise LGTM

Reply:
I think you can get this in terms mode with `functioj.injective.inj_on some.inj` or similar

Reply:
Okay, I've made @ericrbg's change, but I don't think there's much value golfing here. If anyone wants to suggest a working change, or just commit it directly, that's of course fine.